### PR TITLE
🐛 fix: Improve naming convention for Context returning functions

### LINKED ISSUE #3186: 🐛 [Bug]: v3 ctx.Context() is cancelled upon creation
### Bug Description

Running app normally w/ Listen -- context is fine, so I can use it for my db queries.

```go
fmt.Printf("%+v // %+v\n", c.Context(), c.Context().Err())
// #0000000100000001 - 127.0.0.1:8081<->127.0.0.1:48162 - GET http://0.0.0.0:8081/test // <nil>
```
Running a test on the same app -- and context is cancelled upon creation

```go
fmt.Printf("%+v // %+v\n", c.Context(), c.Context().Err())
// #0000000100000001 - 0.0.0.0:0<->0.0.0.0:0 - GET http://example.com/test // context canceled
```
Screenshot from 2024-11-05 00-38-18

### How to Reproduce

1. Create new app
2. Create any handler
3. Print context and it's Err()

See code snippet below

### Expected Behavior

Context in both prod and test should be initialized as not Done, w/o any errors.

### Fiber Version

v3.0.0-beta.3

### Code Snippet (optional)

```go
// main.go

package main

import (
	"fmt"

	"github.com/gofiber/fiber/v3"
)

func main() {
	app := newApp()
	app.Listen(":8081")
}

func newApp() *fiber.App {
	app := fiber.New()
	app.Get("/test", func(c fiber.Ctx) error {
		fmt.Printf("%+v // %+v\n", c.Context(), c.Context().Err())
		return c.SendString("ok")
	})
	return app
}

// main_test.go

package main

import (
	"net/http/httptest"
	"testing"
)

func TestMain(t *testing.T) {
	app := newApp()

	req := httptest.NewRequest("GET", "/test", nil)
	_, err := app.Test(req, -1)
	if err != nil {
		t.Fatal(err)
	}
}
```

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

### PR DESCRIPTION
**Pull request by gaby**
# Description

- The standard practice in golang is to have a function named `Context()` that returns the `context.Context`. In Fiber's case this function was named `UserContext()` which is confusing to users. This PR renames `UserContext()` to `Context()`.
- At the same time we were using `Context()` for returning the FastHTTP `RequestCtx` object. This function has been renamed `RequestCtx()`.

Fixes #3186 

## Changes introduced
- [x] Documentation Update: Detail the updates made to the documentation and links to the changed files.
- [x] API Longevity: Discuss the steps taken to ensure that the new or updated APIs are consistent and not prone to breaking changes.

## Type of change
- [x] Documentation update (changes to documentation)
- [x] Code consistency (non-breaking change which improves code reliability and robustness)

### CODE REVIEW

File: middleware/cache/cache_test.go [hunk 1]
```diff
@@ -894,7 +894,7 @@ func Test_Cache_MaxBytesSizes(t *testing.T) {
 	}))
 
 	app.Get("/*", func(c fiber.Ctx) error {
-		path := c.Context().URI().LastPathSegment()
+		path := c.RequestCtx().URI().LastPathSegment()
```
[the remark this answers is not in the document]
Comment (gaby) [reply middleware/cache/cache_test.go:897]: We are renaming Context() to RequestCtx() and UserContext() to Context()

File: middleware/adaptor/adaptor.go [hunk 2]
```diff
@@ -108,7 +108,7 @@ func HTTPMiddleware(mw func(http.Handler) http.Handler) fiber.Handler {
 					c.Request().Header.Set(key, v)
 				}
 			}
-			CopyContextToFiberContext(r.Context(), c.Context())
+			CopyContextToFiberContext(r.Context(), c.RequestCtx())
```
[the remark this answers is not in the document]
Comment (gaby) [reply middleware/adaptor/adaptor.go:111]: I didn't change the left side. The left side is the net/http context.Context object. The one on the right side is the fasthttp RequestCtx

File: docs/api/ctx.md [hunk 3]
```diff
@@ -2005,18 +2005,18 @@ app.Get("/", func(c fiber.Ctx) error {
 })
 ```
 
-## UserContext
+## Context
```
Comment (ReneWerner87) [inline docs/api/ctx.md:2008]: Don´t forget the documentation for the fasthttp context

File: docs/api/ctx.md [hunk 4]
```diff
@@ -1891,18 +1891,18 @@ app.Get("/", func(c fiber.Ctx) error {
 })
 ```
 
-## SetUserContext
+## SetContext
```
Comment (ReneWerner87) [inline docs/api/ctx.md:1894]: pls add these changes to the "what´s new" section

**Review by ReneWerner87 — CHANGES_REQUESTED** [review]
LGTM, only some minor stuff (markdown documentation)

**Comment by gaby** [comment]
Updating docs in a few minutes.

**Comment by gaby** [comment]
@ReneWerner87 Done

**Comment by JIeJaitt** [comment]
@gaby This PR will be merged in v3. By the way, when will the v3 version be released?

**Comment by JIeJaitt** [comment]
image
Do @gaby need to rename this？

**Review by ReneWerner87 — APPROVED** [review]

**Comment by gaby** [comment]
@JIeJaitt Yes, PR is v3 related. It's a breaking change. Going from v2 to v3 requires changes, it's a major release thus why we are doing the "What's New" document.

The key can stay named the same since its internal to Fiber.

**Review by efectn — APPROVED** [review]

**Pull request merged**